src/status_im/contexts/settings/wallet/saved_addresses/events.cljs

```diff
@@ -23,15 +23,29 @@
 
 (rf/reg-event-fx :wallet/save-address save-address)
 
+(defn- update-saved-addresses
+  [saved-addresses-db new-saved-addresses]
+  (reduce
+   (fn [acc {:keys [address removed? test?] :as saved-address}]
+     (let [db-key (if test? :test :prod)]
+       (if removed?
+         (update acc db-key dissoc address)
+         (assoc-in acc [db-key address] saved-address))))
+   (or saved-addresses-db
+       {:test {}
+        :prod {}})
+   new-saved-addresses))
+
+(defn reconcile-saved-addresses
+  [{:keys [db]} [saved-addresses]]
+  {:db (update-in db [:wallet :saved-addresses] update-saved-addresses saved-addresses)})
+
+(rf/reg-event-fx :wallet/reconcile-saved-addresses reconcile-saved-addresses)
+
 (defn get-saved-addresses-success
-  [{:keys [db]} [raw-saved-addresses]]
-  (let [saved-addresses (reduce
-                         (fn [result {:keys [address test?] :as saved-address}]
-                           (assoc-in result [(if test? :test :prod) address] saved-address))
-                         {:test {}
-                          :prod {}}
-                         (data-store/rpc->saved-addresses raw-saved-addresses))]
-    {:db (assoc-in db [:wallet :saved-addresses] saved-addresses)}))
+  [_ [raw-saved-addresses]]
+  (let [saved-addresses (data-store/rpc->saved-addresses raw-saved-addresses)]
+    {:fx [[:dispatch [:wallet/reconcile-saved-addresses saved-addresses]]]}))
```

Comment:
I'm sorry @seanstrom, do you mean the `on-success` call of event `get-saved-addresses`?

I thought to keep the `:wallet/reconcile-saved-addresses` event free from any data transformations (`data-store/**`) and do the db reconciliation of saved addresses which will help to dispatch this event without any worries in future if needed.

Reply:
Hey @smohamedjavid, yup yup I see what you mean, and I think the separation is good 👍 
The one thing I was mentioning is that we might not need to do an extra dispatch, and we could define `get-saved-addresses-success` sorta like this I think:

```clojure
(defn get-saved-addresses-success
   [cofx [raw-saved-addresses]]
   (let [saved-addresses (data-store/rpc->saved-addresses raw-saved-addresses)]
     {:fx [[:dispatch [:wallet/reconcile-saved-addresses saved-addresses]]]
      :db (:db (reconcile-saved-addresses cofx [saved-addresses]))}))
```

But I suppose this wouldn't be ideal if `reconcile-saved-addresses` would evolve to also produce effects too. So not a blocker for me, but I wanted to mention it could be possible to compose these.